src/udev/udev-event.c

```diff
@@ -881,8 +881,7 @@ static int update_devnode(UdevEvent *event) {
                 return log_device_error_errno(dev, r, "Failed to get devnum: %m");
 
         /* remove/update possible left-over symlinks from old database entry */
-        if (event->dev_db_clone)
-                (void) udev_node_update_old_links(dev, event->dev_db_clone);
+        (void) udev_node_update_old_links(dev, event->dev_db_clone);
```

Comment:
Hmm, why is this check unnecessary? udev_node_update_old_links() needs the args to be nonnull, and I don't see how this is guaranteed here.

Reply:
`dev_db_clone` is always created except for 'remove' uevent. `update_devnode()` is not called on 'remove' uevent. So, it is not necessary.